test/nn/test_pooling.py

```diff
@@ -376,6 +376,23 @@ def test_adaptive_pooling_zero_batch(self, dtype, device):
         mod = torch.nn.AdaptiveAvgPool3d((5, 5, 5)).to(device)
         _test_module_empty_input(self, mod, inp, check_size=False)
 
+    def test_adaptive_pooling_empty_output_size(self):
+        error_msg = "Expected grad_output to have non-zero size for non-batch dimensions"
+
+        for device in ['cpu', 'cuda']:
+            input = torch.rand([1, 64, 10, 9], device=device, requires_grad=True)
+            output_size = 0
+
+            for numel in (2, 3):
```

Comment:
It think `dim` is better name than `numel`.

Reply:
Probably `for dim, pool_type in itertools.product((2, 3), ('max', 'avg'))` would be nicer.

Also, I think it would be nice to test for `1d` implementation of these pooling ops. Wdyt?

Reply:
@kshitij12345, I think there's one more thing added to our offline discussions about adding an error msg to adaptive_{max, avg}_pool1d that this operator does not support the backward call, and I think it's not right to add an error msg to forward mode. (every operator works well for output size zero and will lead to bc-breaking if we change)

I added a test ignoring the name in the above error msg. Is that okay? wdyt?